addons/web_editor/static/src/js/editor/odoo-editor/src/OdooEditor.js

```diff
@@ -2583,6 +2584,131 @@ export class OdooEditor extends EventTarget {
     // TABLE MANAGEMENT
     // ================
 
+    /**
+     * Sets selection in table cells in order to select the cells
+     * rectangularly when pressing shift + arrow key (just like
+     * google docs).
+     * 
+     * @private
+     * @param {KeyboardEvent}
+     */
+    _setSelectionInTable(ev) {
+        const selection = this.document.getSelection();
+        let startTable = closestElement(selection.anchorNode, 'table');
+        let endTable = closestElement(selection.focusNode, 'table');
+        if (!startTable || !endTable || !(startTable.contains(endTable) || endTable.contains(startTable))) {
+            return;
+        }
+        let [startTd, endTd] = [closestElement(selection.anchorNode, 'td'), closestElement(selection.focusNode, 'td')];
+        if (startTable !== endTable) {
```

Comment:
Its a good practice to add docstring while introducing a substantial method, within this method also add docstring explaining what functionalities you have handled where it is required.